micro-rdk-nmea/src/parse_helpers/parsers.rs

```diff
@@ -109,6 +124,20 @@ macro_rules! generate_number_field_readers {
                     }
                 }
             }
+
+            impl FieldReader for NumberFieldWithScale<$t> {
+                type FieldType = f64;
+                fn read_from_cursor(&self, cursor: &mut DataCursor) -> Result<Self::FieldType, NmeaParseError> {
+                    self.number_field.read_from_cursor(cursor).and_then(|x| {
+                        match x {
+                            x if x == <$t>::MAX => { return Err(NumberFieldError::FieldNotPresent("type unavailable".to_string()).into()); },
```

Comment:
do we still want to return a ReadingResponse if the field is absent or is it fine to return a error and no readings even if other values are present?

Reply:
I believe last I checked, gonmea's behavior was to error completely